cvat/apps/engine/ddln/multiannotation.py

```diff
@@ -47,6 +48,58 @@ def request_extra_annotation(task, segments, assignees):
     record_extra_annotation_creation(task, assignments, version)
 
 
+def accept_segments(task, file_path, segments):
+    root_dir = Path(file_path)
+    sequences = [s.sequence_name for s in segments]
+    _move_accepted_sequences(root_dir, sequences)
+    _update_invalid_frames_yaml_file(task, segments, root_dir / "invalid.yaml")
+    _remove_archive_file(file_path)
+
+
+def _move_accepted_sequences(root_dir, sequences):
+    rejected_dir = root_dir / "Rejected_annotation_output"
+    accepted_dir = root_dir / "Annotation_output"
+
+    for sequence_name in sequences:
+        dest_dir = accepted_dir / sequence_name
+        src_dir = rejected_dir / sequence_name
+        dest_dir.mkdir(exist_ok=True)
+
+        for src_file in src_dir.iterdir():
+            dest_file = dest_dir / src_file.relative_to(src_dir)
+            src_file.rename(dest_file)
+
+        src_dir.rmdir()
+
+    if len(list(rejected_dir.iterdir())) == 0:
+        rejected_dir.rmdir()
+
+
+def _update_invalid_frames_yaml_file(task, segments, file_path):
+    """Remove accepted sequences from invalid frames yaml file."""
+    accepted_sequences = {s.sequence_name for s in segments}
+    yaml_writer = DdlnYamlWriter(task.name)
+    seq_name_by_ddln_id = {ddln_id: seq for seq, ddln_id in yaml_writer.id_by_seq_name.items()}
+    rejected_frames = _load_rejected_frames(file_path, seq_name_by_ddln_id)
+    still_rejected_frames = {seq: frames for seq, frames in rejected_frames.items() if seq not in accepted_sequences}
+    yaml_writer.write_invalid_frames(file_path.open('wt'), still_rejected_frames)
```

Comment:
I don't understand why `file_path` is opened for writing in several places throughout the code
shouldn't the YAML be written only once?

Reply:
On annotations export "task_mapping.csv", "ddln.yaml", "invalid.yaml" are open for write once to write their content.

When some of the sequences are accepted manually, I have to update "invalid.yaml" file - that's why I open it once for read, modify it's content in memory, then open it once for write.